src/core/DataFile.cpp

```diff
@@ -901,6 +901,29 @@ void DataFile::upgrade_1_1_91()
 }
 
 
+void DataFile::upgrade_1_2_0_rc3()
+{
+	// Upgrad from earlier bbtrack beat note behaviour of adding
+	// steps if a note is placed after the last step.
+	QDomNodeList list = elementsByTagName( "bbtrack" );
+	for( int i = 0; !list.item( i ).isNull(); ++i )
+	{
+		list = elementsByTagName( "pattern" );
+		for( int i = 0; !list.item( i ).isNull(); ++i )
+		{
+			int patternLength, steps;
+			QDomElement el = list.item( i ).toElement();
+			for( int i = 0; !list.item( i ).isNull(); ++i )
+			{
+				patternLength = el.attribute( "len" ).toInt();
```

Comment:
> If I create some len="" in a .mmp the results are a bit buggy so I think that indicates there probably are no blank "len".

That was my suspicion as well but I wanted to mention it as I'm not sure how they're stored for all use-cases.